File: 04_iteracao_listas.py
# Lista
frutas = ['laranja', 'maçã', 'uva', 'pêra', 'mamão', 'abacate', 'amora']

# Para percorrer uma lista e exibir apenas seus elementos, usamos for..in, como já vimos no arquivo 02
for f in frutas:
    print(f)

print('\n', '-' * 80, sep='')


# Se quisermos percorrer a lista em ordem inversa para exibir apenas seus elementos, podemos usar for..in combinado com reserved()
for f in reversed(frutas):
    print(f)

print('\n', '-' * 80, sep='')


# Agora, se precisarmos exibir, além do elemento, também sua POSIÇÃO, devemos usar range():
for pos in range(len(frutas)):
    print(f'A fruta {frutas[pos]} está na posição {pos}.')

print('\n', '-' * 80, sep='')


# Percurso em ordem inversa usando range() com 3 parâmetros
for i in range(len(frutas) -1, -1, -1):
    print(f'A fruta {frutas[i]} está na posição {i}.')

print('\n', '-' * 80, sep='')

File: js/04_iteracao_listas.js
// Lista
const frutas = ['laranja', 'maçã', 'uva', 'pêra', 'mamão', 'abacate', 'amora'];

// Para percorrer uma lista e exibir apenas seus elementos, usamos um loop for..of
for (const f of frutas) {
    console.log(f);
}

console.log("\n" + "-".repeat(80) + "\n");


// Para percorrer a lista em ordem inversa para exibir apenas seus elementos, usamos um loop for..of em conjunto com o método reverse()
// for (const f of frutas.reverse()) { console.log(f); }
// Porém, o método reverse() modifica o array original

// Para percorrer a lista em ordem inversa para exibir apenas seus elementos sem modificar o array original, podemos fazer uma cópia do array e inverter a cópia
const frutasInvertidas = frutas.slice().reverse();
for (const f of frutasInvertidas) {
    console.log(f);
}

console.log("\n" + "-".repeat(80) + "\n");


// Para percorrer a lista e exibir tanto o elemento quanto sua posição, usamos o método forEach()
frutas.forEach((f, pos) => {
    console.log(`A fruta ${f} está na posição ${pos}.`);
});

console.log("\n" + "-".repeat(80) + "\n");


// Percurso em ordem inversa usando um loop for com 3 parâmetros
for (let i = frutas.length - 1; i >= 0; i--) {
    console.log(`A fruta ${frutas[i]} está na posição ${i}.`);
}

console.log("\n" + "-".repeat(80) + "\n");